### PR TITLE
BeamSpot DQM clients modified to log to onlineDB only when beam status from TCDS is STABLE (backport for 12_3_X)

### PR DESCRIPTION
**Pull request by dzuolo**
Following discussion with @ggovi BeamSpot DQM clients have been modified in such a way that they log to onlineDB only when the beam mode from TCDS is STABLE. This prevents the generation of dummy logs containing only "fit failed" reports during, e.g., cosmics data taking.

Code in this PR compile and the following local tests were successful

-scram b code-format
-scram b code-checks
-scramv1 b runtests

This PR is a backport of https://github.com/cms-sw/cmssw/pull/37614

FYI: @gennai @francescobrivio

### CODE REVIEW

**Comment by dzuolo** [comment]
test parameters:

- addpkg = DQM/Integration

**Comment by dzuolo** [comment]
Apparently I am not an "authorised user" that can issue the test command so if somebody could it for me I would appreciate it :)

**Comment by mmusich** [comment]
test parameters:
   * addpkg = DQM/Integration

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@dzuolo the title is clipped (it ends in ...) and also it doesn't say it's for 12.3.X

File: DQM/BeamMonitor/plugins/BeamMonitor.cc [hunk 1]
```diff
@@ -661,6 +664,12 @@ void BeamMonitor::beginLuminosityBlock(const LuminosityBlock& lumiSeg, const Eve
 
 // ----------------------------------------------------------
 void BeamMonitor::analyze(const Event& iEvent, const EventSetup& iSetup) {
+  Handle<TCDSRecord> tcdsData;
+  iEvent.getByToken(tcdsToken_, tcdsData);
```
Comment (mmusich) [inline DQM/BeamMonitor/plugins/BeamMonitor.cc:668]: same comment as https://github.com/cms-sw/cmssw/pull/37614#discussion_r852887001

**Comment by francescobrivio** [comment]
backport of #37614

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by francescobrivio** [comment]
@jfernan2 @ahmad3213 this should also be tested in the playback system in P5 please.

**Comment by dzuolo** [comment]
The log file from the 140.56 is not really explicative https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-90f9f0/24045/runTheMatrixINPUT-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18/step2_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18.log. It seems that the file that was opened has not been closed, like the process was interrupted for some reason, since the printouts are identical to the ones from https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-90f9f0/24045/runTheMatrixINPUT-results/140.57_RunHI2018Reduced+RunHI2018Reduced+RECOHID18+HARVESTDHI18/step2_RunHI2018Reduced+RunHI2018Reduced+RECOHID18+HARVESTDHI18.log

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by francescobrivio** [comment]
> @jfernan2 @ahmad3213 this should also be tested in the playback system in P5 please.

also tagging @pmandrik

**Comment by francescobrivio** [comment]
> @francescobrivio this PR was tested sucessfully by @pmandrik at P5 this morning. 

Ah that's great news! Then once the tests are done we are ready to approve this (provided jenkins tests are succesful of course 😄 )

> I pass the baton to @emanueleusai in what approval concerns since he has kindly agreed to replace me on the PR review. Thanks!

Thank you very much for your help Javi! ❤️

**Comment by emanueleusai** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**